src/apps/explorer/pages/AppData/config.tsx

```diff
@@ -0,0 +1,290 @@
+  'Type error',
+  "Failed to execute 'setRequestHeader' on 'XMLHttpRequest': String contains non ISO-8859-1 code point.",
+]
+
+export type FormProps = Record<string, any>
+
+export const getSchema = async (): Promise<JSONSchema7> => {
+  const latestSchema = (await getAppDataSchema(LATEST_APP_DATA_VERSION)).default as JSONSchema7
+  deleteAllPropertiesByName(latestSchema, 'examples')
+  deleteAllPropertiesByName(latestSchema, '$id')
+  return formatSchema(latestSchema)
+}
+
+const formatSchema = (schema: JSONSchema7): JSONSchema7 => {
+  const formattedSchema = structuredClone(schema)
+
+  if (formattedSchema?.properties?.version) {
+    formattedSchema.properties.version['readOnly'] = true
+    formattedSchema.properties.version['default'] = LATEST_APP_DATA_VERSION
+  }
+
+  if (formattedSchema?.properties?.metadata['properties'].quote) {
+    const requiredFields = formattedSchema.properties.metadata['properties'].quote.required
+    deletePropertyPath(formattedSchema, 'properties.metadata.properties.quote.required')
+
+    formattedSchema.properties.metadata['properties'].quote.properties.version['readOnly'] = true
+    formattedSchema.properties.metadata['properties'].quote.properties.version['default'] =
+      LATEST_QUOTE_METADATA_VERSION
+
+    formattedSchema.properties.version['default'] = LATEST_APP_DATA_VERSION
+    const quoteProperties = formattedSchema.properties.metadata['properties'].quote.properties
+    formattedSchema.properties.metadata['properties'].quote.properties = {
+      enableQuote: { type: 'boolean', title: 'Enable/Disable' },
+    }
+    quoteDependencies.enableQuote.oneOf[0] = {
+      properties: {
+        ...quoteDependencies.enableQuote.oneOf[0].properties,
+        ...quoteProperties,
+      },
+      required: requiredFields,
+    }
+    formattedSchema.properties.metadata['properties'].quote.dependencies = quoteDependencies
+  }
+
+  if (formattedSchema?.properties?.metadata['properties'].referrer) {
+    const requiredFields = formattedSchema.properties.metadata['properties'].referrer.required
+    deletePropertyPath(formattedSchema, 'properties.metadata.properties.referrer.required')
+
+    formattedSchema.properties.metadata['properties'].referrer.properties.version['readOnly'] = true
+    formattedSchema.properties.metadata['properties'].referrer.properties.version['default'] =
+      LATEST_REFERRER_METADATA_VERSION
+    const referrerProperties = formattedSchema.properties.metadata['properties'].referrer.properties
+    formattedSchema.properties.metadata['properties'].referrer.properties = {
+      enableReferrer: { type: 'boolean', title: 'Enable/Disable' },
+    }
+    referrerDependencies.enableReferrer.oneOf[0] = {
+      properties: {
+        ...referrerDependencies.enableReferrer.oneOf[0].properties,
+        ...referrerProperties,
+      },
+      required: requiredFields,
+    }
+    formattedSchema.properties.metadata['properties'].referrer.dependencies = referrerDependencies
+  }
```

Comment:
Lines 46 to 88 are pretty similar.
Can you extract a function to deal with them?